framework/model/pkg/leaderelection/kubeleaderelector.go

```diff
@@ -0,0 +1,137 @@
+package leaderelection
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+var _ LeaderElector = &KubeLeaderElector{}
+
+// KubeLeaderElector implements LeaderElector based on client-go leaderelection.
+// The old leader has a few seconds to quit tasks before switching to the new
+// leader. Tasks that do not exit in time can lead to the fact that there are
+// two leaders at the same time.
+// Furthermore, this implementation does not guarantee that there is only one
+// leader. More information can be obtained from:
+//
+//	https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go
+//  
+// NOTE:
+// KubeLeaderElector does not strictly implement the LeaderElector, and
+// the behavior differences are as follows: 
+//   1. The onStartedLeading callbacks are packaged and called asynchronously,
+//      and executed synchronously in order in the asynchronous logic.
+//   2. The onStoppedLeading callbacks will be executed when exiting,
+//      even without being a leader.
+type KubeLeaderElector struct {
+	startCbLock               sync.RWMutex
+	onStartedLeadingCallbacks []func(context.Context)
+	stopCbLock                sync.RWMutex
+	onStopLeadingCallbacks    []func()
+
+	id        string
+	name      string
+	namespace string
+
+	// config is the rest.config used to talk to the apiserver.
+	config *rest.Config
+
+	le *leaderelection.LeaderElector
+}
+
+func NewKubeLeaderElector(config *rest.Config, namespace, name string) *KubeLeaderElector {
+	workload, _ := os.Hostname()
+	id := workload + "_" + uuid.New().String()
+	return &KubeLeaderElector{
+		id:        id,
```

Comment:
hostname + uuid? perhaps it will cause each instance has a different id and election not work.